pandas/core/computation/parsing.py

```diff
@@ -168,6 +177,87 @@ def tokenize_backtick_quoted_string(
     return BACKTICK_QUOTED_STRING, source[string_start:string_end]
 
 
+class ParseState(Enum):
+    DEFAULT = 0
+    IN_BACKTICK = 1
+    IN_SINGLE_QUOTE = 2
+    IN_DOUBLE_QUOTE = 3
+
+
+def _split_by_backtick(s: str) -> list[tuple[bool, str]]:
+    """
+    Splits a str into substrings along backtick characters (`).
+
+    Disregards backticks inside quotes.
+
+    Parameters
+    ----------
+    s : str
+        The Python source code string.
+
+    Returns
+    -------
+    substrings: list[tuple[bool, str]]
+        List of tuples, where each tuple has two elements:
+        The first is a boolean indicating if the substring is backtick-quoted.
+        The second is the actual substring.
+    """
+    substrings = []
+    substr = ""
+    i = 0
+    parse_state = ParseState.DEFAULT
+    while i < len(s):
+        char = s[i]
+
+        match char:
+            case "`":
+                # start of a backtick-quoted string
+                if parse_state == ParseState.DEFAULT:
+                    if substr:
+                        substrings.append((False, substr))
+                    substr = char
+                    i += 1
```

Comment:
I think its less logic to maintain if you just always let the loop do the increment and break within the branches, instead of using continue.

Is the issue here that you want to do `substr = char` and cannot do the `substr += char` that happens at the end of the loop iteration?

Reply:
Yea...the logic for the three cases here are different from the `substr.append(char); i += 1` logic at the end.
I also like being able to compare at a glance how the logic for the start of a backtick-quoted string differs from the logic for the end of the backtick-quoted string.
Not strongly attached tho, lmk what you think.

Reply:
Sounds good - let's roll with it. Can always be refactored later if needed